src/client/datascience/jupyter/kernels/kernel.ts

```diff
@@ -224,23 +225,55 @@ export class Kernel implements IKernel {
     public async start(options?: { disableUI?: boolean }): Promise<void> {
         await this.startNotebook(options);
     }
-    public async interrupt(): Promise<InterruptResult> {
+    public async interrupt(): Promise<void> {
```

Comment:
This way the UI for interrupt & restart is in one place.
E.g. we have `restart` being called internally as well.

Reply:
& the progress messages & the like are not displayed.